### PR TITLE
feat: add metrics for highest pruned block per segment

### PR DESCRIPTION
**Pull request by Paulius0112**
Pull request for issue https://github.com/paradigmxyz/reth/issues/4921

Adding prune metric for highest pruned block per segment

Example of metric:
`# TYPE reth_pruner_segments_highest_pruned_block gauge
reth_pruner_segments_highest_pruned_block{segment="TransactionLookup"} 19257070
reth_pruner_segments_highest_pruned_block{segment="Receipts"} 1919999
reth_pruner_segments_highest_pruned_block{segment="SenderRecovery"} 19157070
`

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]
lgtm

**Review by rkrasiuk — APPROVED** [review]
lgtm pending fmt fix

File: crates/prune/src/metrics.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use reth_metrics::{metrics::Histogram, Metrics};
+use reth_metrics::{metrics::Histogram, metrics::Gauge, Metrics};
```
Comment (rkrasiuk) [inline crates/prune/src/metrics.rs:1]:
```suggestion
use reth_metrics::{
    metrics::{Gauge, Histogram},
    Metrics,
};
```

File: crates/prune/src/metrics.rs [hunk 2]
```diff
@@ -29,4 +29,6 @@ impl Metrics {
 pub(crate) struct PrunerSegmentMetrics {
     /// Pruning duration for this segment
     pub(crate) duration_seconds: Histogram,
+    #[metric(rename = "highest_pruned_block", describe = "Highest pruned block per segment")]
+    pub(crate) highest_pruned_block: Gauge,
```
Comment (rkrasiuk) [inline crates/prune/src/metrics.rs:33]:
```suggestion
    /// Highest pruned block per segment
    pub(crate) highest_pruned_block: Gauge,
```

**Pull request merged**